pkg/collector/corechecks/snmp/checkconfig/profile.go

```diff
@@ -154,6 +165,27 @@ func recursivelyExpandBaseProfiles(definition *profileDefinition, extends []stri
 		}
 		definition.Metrics = append(definition.Metrics, baseDefinition.Metrics...)
 		definition.MetricTags = append(definition.MetricTags, baseDefinition.MetricTags...)
+		for baseResName, baseResource := range baseDefinition.Metadata {
```

Comment:
This part seems sensitive, can we unit test it?

Reply:
Extracted merging profile definition into a new method and unit tested it here e1bb9a0e1